File: ruby/test/wire/evaluation_criteria_test.rb
# frozen_string_literal: true

require_relative "wiremock_test_case"

class EvaluationCriteriaWireTest < WireMockTestCase
  def setup
    super

    @client = Sikaru::Client.new(
      api_key: "<token>",
      base_url: WIREMOCK_BASE_URL
    )
  end

  def test_evaluation_criteria_list_criteria_with_wiremock
    test_id = "evaluation_criteria.list_criteria.0"

    @client.evaluation_criteria.list_criteria(
      project_id: "project_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "evaluation_criteria.list_criteria.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/projects/project_id/evaluation-criteria",
      query_params: nil,
      expected: 1
    )
  end
end

File: rust/tests/evaluation_criteria_test.rs
use sikaru::prelude::*;

mod wire_test_utils;

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_evaluation_criteria_list_criteria_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = Sikaru::new(config).expect("Failed to build client");

    let result = client
        .evaluation_criteria
        .list_criteria(
            &"project_id".to_string(),
            &ListCriteriaQueryRequest {
                ..Default::default()
            },
            None,
        )
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count(
        "GET",
        "/v1/projects/project_id/evaluation-criteria",
        None,
        1,
    )
    .await
    .unwrap();
}
